src/arrayMethodJoin.js

```diff
@@ -4,8 +4,24 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let result = '';
+
+    if (this.length === 0) {
+      return '';
+    }
+
+    for (let i = 0; i < this.length - 1; i++) {
+      if (this[i] === null || this[i] === undefined) {
+        result += '';
+      } else {
+        result += this[i];
+      }
```

Comment:
```suggestion
      if (this[i] !== null || this[i] !== undefined) {
        result += this[i];
      }
```
You can swap your expressions in `if-else` by places and change the check from `===` to `!==`. In this case, you will be checking if the element is not null or undefined, and if so - add it to the string. If it's null or undefined - nothing will happen in that case.
And this wil help you to get rid of unnecessary `else` statement